### PR TITLE
Refactor TFVARS to ENV VARs and create new s3 module

### PR DESCRIPTION
**Pull request by mabe13**
This PR moves the `tfvars` for the instructure Terraform to enviroment variables. It also creates a new S3 module to replace the current one which has become outdate, with one base of the a module support by Terraform.

Note : current only a draft pull request as it has passed a `terraform plan `check, but changes need to check with `terraform apply`

### CODE REVIEW

**Comment by mabe13** [comment]
Note currently has two s3 terraform modules `s3` and `s3_new` once were happy with the new module we can remove `s3` and rename `s3_new` to s3

**Comment by stefaniuk** [comment]
Code review comments:
- Need to create the `dev` profile as `nonprod` is the account and shouldn't be used for naming profiles
- Remove `infrastructure/stacks/tfvars/nonprod.tfvars` and `infrastructure/stacks/tfvars/prod.tfvars`
- Only use the `Service` tag for infrastructure, drop the others
- Derive `log_bucket` from `name` and `log_prefix` from `service_prefix`
- As an example check this file https://github.com/nhsd-ddce/make-devops/blob/master/build/automation/lib/terraform/template/stacks/rds/variables.tf to see what sort of default variables we will be pasing to that module
- Do not amend `build/automation/var/platform-texas/account-live-k8s-nonprod.mk` and `build/automation/var/platform-texas/account-live-k8s-prod.mk` files and move the variables to either `project.mk` or profile file
- Do not name variables using the following pattern `TF_VAR_s3_pu_data_load_process_bucket` but use e.g. `SERVICE_BUCKET` 
- Remove `infrastructure/Makefile` file

File: build/automation/var/profile/live.mk [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+K8S_APP_NAMESPACE := $(K8S_APP_NAMESPACE)-prod
+K8S_JOB_NAMESPACE := $(K8S_JOB_NAMESPACE)-prod
+
+# ==============================================================================
+# Service variables
+
+# ==============================================================================
+# Infrastructure variables
+
+AWS_ENV = prod
```
Comment (stefaniuk) [inline build/automation/var/profile/live.mk:13]: There is already a variable `AWS_ACCOUNT_NAME` for it.

File: build/automation/var/profile/dev.mk [hunk 2]
```diff
@@ -6,6 +6,4 @@
 # ==============================================================================
 # Infrastructure variables
 
-STACKS = service
-
-TF_VAR_profile = $(PROFILE)
+AWS_ENV = nonprod
```
Comment (stefaniuk) [inline build/automation/var/profile/dev.mk:9]: There is already a variable `AWS_ACCOUNT_NAME` for it.

File: build/automation/var/platform-texas/platform-texas.mk [hunk 3]
```diff
@@ -7,17 +7,20 @@ AWS_ROLE_SESSION = jenkins
 AWS_ROLE := $(if $(HUDSON_URL),$(AWS_ROLE_JENKINS),Developer)
```
Comment (stefaniuk) [inline build/automation/var/platform-texas/platform-texas.mk:7]: I can see that `TEXAS_NHSD_PREFIX` has been added, Any reason why we need to change this file?

File: build/automation/var/profile/local.mk [hunk 4]
```diff
@@ -3,9 +3,13 @@
...
 DB_URL := localhost
 DB_SECRET_NAME := pu-data-load-process/deployment
 AWS_S3_BUCKET_NAME := pu-data-load-process-bucket
 AWS_SQS_QUEUE_URL := pu-data-load-process-queue.fifo
 IS_PROD_DEPLOYMENT := false
+
+# ==============================================================================
+# Infrastructure variables
+
+AWS_ENV = nonprod
```
Comment (stefaniuk) [inline build/automation/var/profile/local.mk:15]: There is already a variable AWS_ACCOUNT_NAME for it.

Comment (mabe13) [reply build/automation/var/profile/local.mk]: removed

File: build/automation/var/platform-texas/platform-texas.mk (hunk 3 above)
Comment (mabe13) [reply build/automation/var/platform-texas/platform-texas.mk:7]: removed

File: build/automation/var/profile/dev.mk (hunk 2 above)
Comment (mabe13) [reply build/automation/var/profile/dev.mk]: removed

File: build/automation/var/profile/live.mk (hunk 1 above)
Comment (mabe13) [reply build/automation/var/profile/live.mk]: removed

**Comment by mabe13** [comment]
@stefaniuk This pull is request ready to be reviewed with the latest and hopefully last changes  🤞

**Pull request merged**